synapse/logging/context.py

```diff
@@ -761,18 +761,21 @@ def run_in_background(  # type: ignore[misc]
 
 @overload
 def run_in_background(
-    f: Callable[..., R], *args: Any, **kwargs: Any
+    f: Callable[P, R], *args: P.args, **kwargs: P.kwargs
 ) -> "defer.Deferred[R]":
     ...
 
 
-def run_in_background(
+def run_in_background(  # type: ignore[misc]
+    # The `type: ignore[misc]` above suppresses
+    # "Overloaded function implementation does not accept all possible arguments of signature 1"
+    # "Overloaded function implementation does not accept all possible arguments of signature 2"
```

Comment:
I'm not quite sure I understand the issue -- the `Union` below seems to capture both potential cases?

Reply:
I don't understand the error myself. I assumed it was just mypy being buggy.

Reply:
Thanks!